File: tests/refs/binary_search.py
"""Reference: classical iterative binary search on a sorted array.

Returns the index of `target` (any matching index — the leftmost in the
fixture cases) or -1 if not found.
"""

from __future__ import annotations
from typing import List


def binarySearch(arr: List[int], target: int) -> int:
    low, high = 0, len(arr) - 1
    while low <= high:
        mid = low + (high - low) // 2
        if arr[mid] == target:
            return mid
        elif arr[mid] < target:
            low = mid + 1
        else:
            high = mid - 1
    return -1

File: tests/refs/BinarySearch.java
/*
 * Reference Java implementation for tests/cases/binary_search.json.
 * Iterative binary search; returns matched index or -1.
 */
public class BinarySearch {
    public static long binarySearch(long[] arr, long target) {
        int low = 0, high = arr.length - 1;
        while (low <= high) {
            int mid = low + (high - low) / 2;
            if (arr[mid] == target) return mid;
            else if (arr[mid] < target) low = mid + 1;
            else high = mid - 1;
        }
        return -1;
    }
}
